.github/workflows/cryostat-build.yml

```diff
@@ -0,0 +1,47 @@
+on:
+  workflow_call:
+    inputs:
+      build-arch:
+        required: true
+        type: string
+      core-version:
+        required: true
+        type: string
+
+jobs:
+  build-image:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install qemu
+      if: ${{ inputs.build-arch != 'amd64' }}
+      continue-on-error: true
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y qemu-user-static
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ github.repository_owner }}/cryostat
+        ref: main
+        submodules: true
+        fetch-depth: 0
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: maven
```

Comment:
https://github.com/actions/setup-java#caching-packages-dependencies

Sadly, cache key used hash of "pom.xml" which won't work for our case. Have to use action/cache directly.

Tho, Can we use these configs to set up maven login?

https://github.com/actions/setup-java#maven-options